### PR TITLE
MM-68261 Add Button component to shared package

### PR DESCRIPTION
**Pull request by hmhealey**
#### Summary
Most places in the app where we use buttons use the same CSS classes from Bootstrap (`btn` along with variations like `btn-primary` or `btn-danger`). This PR consolidates over 60% of those into a new `Button` component. The remaining 40% are one of the following:
- A button using non-standard CSS - They may still duplicate the standard CSS, but I didn't change most of that because I wanted to limit the risks of this causing user-visible changes
- Icon buttons using `btn-icon` - These have similar designs to regular buttons, but I think it's cleaner to give them their own shared component.
- Links that appear to be buttons - I've considered giving these their own shared component as well, but https://github.com/mattermost/mattermost/pull/36328 is my current solution for that.

For the `Button`'s API, I went with passing most props directly through to the component to make it as easy as possible to use this component unlike Daniel's previous PR to do this. I'm still not certain this is the best approach in the long run, but I'm happy with the result at the moment. The only one I'm considering changing further is renaming the `className` prop to `extraClass` to prevent people from accidentally using it instead of the other purpose-built props that match the design system.

While there's some changes here that require updating intermediate components (like `SpinnerButton`, `ConfirmModal`, or `GenericModal`), most of the changes in this PR look the following:
```jsx
// Before
<button className='btn btn-primary btn-lg btn-danger my-custom-button'/>

// After
<Button
    emphasis='primary'
    size='lg'
    variant='destructive'
    className='my-custom-button' // Ideally, this shouldn't be needed and it might be worth followup to get rid of this wherever possible
/>
```

There are some small functional changes here, mostly due to the few places where I did get rid of some non-standard button CSS. I also made a couple accessibility fixes as well, but I avoided any significant ones.

#### Ticket Link
MM-68261

#### Release Note
```release-note
Standardized many buttons throughout the app which may result in minor UX changes
```

## Change Impact: 🟡 Medium

**Regression Risk:** The change introduces a new shared Button component and replaces hundreds of native/button-class usages across many UI modules and shared modal primitives; this large surface area and removal of legacy CSS/class props increase risk of styling, focus/keyboard, and subtle behavioral regressions (especially for icon-only buttons, link-as-button patterns, and modal confirm/cancel flows). A few TypeScript prop signature changes and removed CSS rules further raise the chance of uncovered edge cases despite targeted test updates.

**QA Recommendation:** Full manual QA with visual-regression checks is recommended, focusing on button states (disabled/loading/variant), keyboard/focus accessibility, modal confirm/cancel behavior (ConfirmModal/GenericModal/GenericModalRedux), and critical admin and dialog flows; run E2E suites and smoke tests across high-usage areas to catch regressions quickly.

*Generated by CodeRabbitAI*

### CODE REVIEW

File: webapp/channels/src/components/admin_console/group_settings/groups_list/groups_list.tsx [hunk 1]
```diff
@@ -183,21 +185,21 @@ export default class GroupsList extends React.PureComponent<Props, State> {
...
             );
         default:
             return (
-                <button
+                <Button
                     type='button'
-                    className='btn btn-primary'
+                    emphasis='primary'
                     disabled={this.props.readOnly}
                 >
```
[the remark this answers is not in the document]
Comment (hmhealey) [reply webapp/channels/src/components/admin_console/group_settings/groups_list/groups_list.tsx:196]: Changing functional behaviour is out of scope for this PR

File: webapp/channels/src/components/announcement_bar/renewal_link/renewal_link.tsx [hunk 2]
```diff
@@ -28,20 +29,17 @@ const RenewalLink = (props: RenewalLinkProps) => {
...
-    const buttonClassName = props.className || defaultClassName;
-
     return (
-        <button
-            className={buttonClassName}
+        <Button
+            emphasis='tertiary'
+            size='xs'
+            variant='inverted'
+            className='annnouncementBar__renewLicense'
```
[the remark this answers is not in the document]
Comment (hmhealey) [reply webapp/channels/src/components/announcement_bar/renewal_link/renewal_link.tsx:37]: This spelling matches the CSS. I'm not fixing that typo in this PR

File: webapp/channels/src/components/commercial_support_modal/commercial_support_modal.tsx [hunk 3]
```diff
@@ -227,17 +228,17 @@ export default class CommercialSupportModal extends React.PureComponent<Props, S
...
                                 </div>
                             )}
-                            <a
-                                className='btn btn-primary DownloadSupportPacket'
+                            <Button
+                                emphasis='primary'
+                                className='DownloadSupportPacket'
                                 onClick={this.downloadSupportPacket}
-                                rel='noopener noreferrer'
                             >
```
[the remark this answers is not in the document]
Comment (hmhealey) [reply webapp/channels/src/components/commercial_support_modal/commercial_support_modal.tsx:235]: That's a functional issue in the existing code. I'm not fixing those in this PR.

File: webapp/channels/src/components/widgets/admin_console/admin_panel_with_button.tsx [hunk 4]
```diff
@@ -33,15 +35,16 @@ const AdminPanelWithButton = ({
...
         const buttonId = (buttonText.defaultMessage as string || '').split(' ').join('-').toLowerCase();
         button = (
-            <a
-                className={classNames('btn', 'btn-primary', {disabled})}
-                onClick={disabled ? (e) => e.preventDefault() : onButtonClick}
+            <Button
+                emphasis='primary'
+                className={classNames({disabled})}
+                onClick={disabled ? undefined : onButtonClick}
                 data-testid={buttonId}
```
[the remark this answers is not in the document]
Comment (hmhealey) [reply webapp/channels/src/components/widgets/admin_console/admin_panel_with_button.tsx:42]: That's a functional issue in the existing code. I'm not fixing those in this PR.

File: webapp/platform/shared/src/components/button/button.tsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    return (
+        <button
+            ref={ref}
+            className={classNames('btn', emphasisClass, sizeClass, variantClass, /*{
+                // 'btn-full': width === 'full',
+            },*/ className)}
+            {...otherProps}
+        >
```
[the remark this answers is not in the document]
Comment (hmhealey) [reply webapp/platform/shared/src/components/button/button.tsx:77]: We can do that in a followup since it could be a functional change

File: e2e-tests/cypress/tests/integration/channels/move_thread/move_thread_from_dm_spec.js [hunk 6]
```diff
@@ -207,10 +207,10 @@ describe('Move Thread', () => {
 
             if (cancel) {
                 // * Assert if button is active
-                cy.get('.MoveThreadModal__cancel-button').should('not.be.disabled').type('{esc}', {force: true});
+                cy.findByText('Cancel').should('not.be.disabled').click();
             } else {
                 // * Assert if button is active
-                cy.get('.GenericModal__button.confirm').should('not.be.disabled').type('{enter}', {force: true});
+                cy.findByText('Move').should('not.be.disabled').click();
             }
```
[the remark this answers is not in the document]
Comment (hmhealey) [reply e2e-tests/cypress/tests/integration/channels/move_thread/move_thread_from_dm_spec.js:214]: Done

File: e2e-tests/cypress/tests/integration/channels/move_thread/move_thread_from_private_channel_spec.js [hunk 7]
```diff
@@ -198,10 +198,10 @@ describe('Move thread', () => {
 
             if (cancel) {
                 // * Assert if button is active
-                cy.get('.MoveThreadModal__cancel-button').should('not.be.disabled').click();
+                cy.findByText('Cancel').should('not.be.disabled').click();
             } else {
                 // * Assert if button is active
-                cy.get('.GenericModal__button.confirm').should('not.be.disabled').click();
+                cy.findByText('Move').should('not.be.disabled').click();
             }
```
[the remark this answers is not in the document]
Comment (hmhealey) [reply e2e-tests/cypress/tests/integration/channels/move_thread/move_thread_from_private_channel_spec.js:205]: Done

File: webapp/channels/src/components/admin_console/admin_button_outline/admin_button_outline.tsx
Comment (hmhealey) [inline webapp/channels/src/components/admin_console/admin_button_outline/admin_button_outline.tsx:1]: This component isn't actually used

File: webapp/channels/src/components/admin_console/plugin_management/plugin_management.tsx [hunk 8]
```diff
@@ -1155,17 +1150,17 @@ export class PluginManagement extends OLDAdminSettings<Props, State> {
                                     label={<FormattedMessage {...messages.uploadTitle}/>}
                                 >
                                     <div className='file__upload'>
-                                        <button
+                                        <Button
                                             type='button'
-                                            className={classNames(['btn', {'btn-tertiary': enableUploads}])}
+                                            emphasis='tertiary'
```
Comment (hmhealey) [inline webapp/channels/src/components/admin_console/plugin_management/plugin_management.tsx:1155]: The old conditional isn't needed because the disabled styling overrides `btn-tertiary`

File: webapp/channels/src/components/admin_console/access_control/policy_details/policy_details.tsx [hunk 9]
```diff
@@ -673,8 +673,7 @@ function PolicyDetails({
                             defaultMessage='Delete Policy'
                         />
                     }
-                    confirmButtonClassName='btn btn-danger'
-                    isDeleteModal={true}
```
Comment (hmhealey) [inline webapp/channels/src/components/admin_console/access_control/policy_details/policy_details.tsx:677]: `isDeleteModal` is redundant since it was always true whenever we rendered a `btn-danger` button, so I changed the logic relying on it to use `confirmButtonVariant`

File: webapp/channels/src/components/admin_console/team_channel_settings/channel/details/channel_profile.tsx [hunk 10]
```diff
@@ -79,23 +79,16 @@ export const ChannelProfile = (props: ChannelProfileProps): JSX.Element => {
...
                     <div className='AdminChannelDetails_archiveContainer'>
-                        <button
+                        <Button
                             type='button'
-                            className={
-                                classNames(
-                                    'btn',
-                                    'btn-secondary',
-                                    {'btn-danger': !isArchived},
-                                    {disabled: isDisabled},
```
Comment (hmhealey) [inline webapp/channels/src/components/admin_console/team_channel_settings/channel/details/channel_profile.tsx:89]: There's a few places where we used `disabled` CSS classes instead of actually disabling the button. I tried to remove that wherever there didn't seem to be actual styles or tests relying on it

File: webapp/channels/src/components/admin_console/team_channel_settings/team/details/team_profile.tsx
Comment (hmhealey) [inline webapp/channels/src/components/admin_console/team_channel_settings/team/details/team_profile.tsx:1]: This is one of the only places where I aggressively removed redundant CSS and got rid of a `disabled` class. I made sure to manually test these though

File: webapp/channels/src/components/admin_console/team_channel_settings/abstract_list.tsx [hunk 11]
```diff
@@ -154,23 +155,25 @@ export default class AbstractList extends React.PureComponent<Props, State> {
                             }}
                         />
                     </div>
-                    <button
+                    <Button
                         type='button'
-                        className={'btn btn-tertiary prev ' + (firstPage ? 'disabled' : '')}
```
Comment (hmhealey) [inline webapp/channels/src/components/admin_console/team_channel_settings/abstract_list.tsx:159]: The `prev` and `next` classes didn't seem to have any attached CSS here. I generally tried to remove CSS classes that had no styles attached and left ones which had them attached

File: webapp/channels/src/components/advanced_text_editor/rewrite_menu.tsx [hunk 12]
```diff
@@ -156,8 +157,9 @@ export default function RewriteMenu({
                                 id='texteditor.rewrite.rewriting'
                                 defaultMessage='Rewriting'
                             />
-                            <button
-                                className='btn btn-xs'
+                            <Button
+                                emphasis='tertiary'
```
Comment (hmhealey) [inline webapp/channels/src/components/advanced_text_editor/rewrite_menu.tsx:161]: This is a functional change because we don't have a UX design that maps to Bootstrap's default button formatting. It looks very similar to a grey version of the tertiary button, so I went with that

File: webapp/channels/src/components/announcement_bar/purchase_link/purchase_link.tsx [hunk 13]
```diff
@@ -3,14 +3,15 @@
...
+import {Button} from '@mattermost/shared/components/button';
+
 import useOpenSalesLink from 'components/common/hooks/useOpenSalesLink';
 
 import './purchase_link.scss';
 
 export interface Props {
     buttonTextElement: JSX.Element;
     eventID?: string;
-    className?: string;
```
Comment (hmhealey) [inline webapp/channels/src/components/announcement_bar/purchase_link/purchase_link.tsx:13]: This prop was always used with the same value, so I inlined it

File: webapp/channels/src/components/browse_channels/browse_channels.scss [hunk 14]
```diff
@@ -276,10 +276,6 @@
     .secondary-message {
         margin-bottom: 30px;
     }
-
-    #createNewChannelButton {
```
Comment (hmhealey) [inline webapp/channels/src/components/browse_channels/browse_channels.scss:280]: This matches the regular button CSS

File: webapp/channels/src/sass/components/_groups.scss
Comment (hmhealey) [inline webapp/channels/src/sass/components/_groups.scss:1]: These are more duplicated styles. I think those colours might be slightly different than the standard ones, but I think these matched the standard theme colours before those were updated at some point

File: webapp/platform/shared/src/components/button/button.tsx [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+export type ButtonVariant = '' | 'destructive' | 'inverted';
+
+export interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+    children: React.ReactNode;
+
+    emphasis?: ButtonEmphasis;
+    size?: ButtonSize;
+    variant?: ButtonVariant;
+
+    // width?: 'full' | number; // TODO
```
Comment (hmhealey) [inline webapp/platform/shared/src/components/button/button.tsx]: I'll add this in a followup since it's not used in very many places and I don't want to add more to this PR if I can avoid this

File: webapp/channels/src/components/__snapshots__/setting_picture.test.tsx.snap [hunk 16]
```diff
@@ -205,7 +205,7 @@ exports[`components/SettingItemMin should match snapshot, on loading picture 1`]
             </button>
             <button
               aria-label="Uploading..."
-              class="btn btn-inactive disabled"
+              class="btn btn-primary"
```
Comment (larkox) [inline webapp/channels/src/components/__snapshots__/setting_picture.test.tsx.snap:208]: This looks a bit off, but maybe intended.

It used to be a `btn-inactive` and now is just `btn-primary`.

File: webapp/channels/src/components/admin_console/group_settings/groups_list/__snapshots__/groups_list.test.tsx.snap [hunk 17]
```diff
@@ -471,7 +471,7 @@ exports[`components/admin_console/group_settings/GroupsList.tsx should match sna
         1 - 2 of 2
       </div>
       <button
-        class="btn btn-tertiary prev disabled"
+        class="btn btn-tertiary disabled"
```
Comment (larkox) [inline webapp/channels/src/components/admin_console/group_settings/groups_list/__snapshots__/groups_list.test.tsx.snap:474]: This may not be intended. You are missing lots of prev and next classes here, due to the changes in the code file.

File: webapp/channels/src/components/admin_console/license_settings/enterprise_edition/enterprise_edition_left_panel.tsx [hunk 18]
```diff
@@ -159,8 +162,8 @@ const EnterpriseEditionLeftPanel = ({
                             })}
                             disabled={!isLicenseSetByEnvVar}
                         >
-                            <button
-                                className='btn btn-primary upload-license-btn'
+                            <Button
+                                emphasis='primary'
```
Comment (larkox) [inline webapp/channels/src/components/admin_console/license_settings/enterprise_edition/enterprise_edition_left_panel.tsx:166]: 0/5 whether intended or not, but you are losing the upload-license-btn class.

File: webapp/channels/src/components/admin_console/license_settings/enterprise_edition/enterprise_edition_left_panel.tsx [hunk 19]
```diff
@@ -217,15 +220,16 @@ const EnterpriseEditionLeftPanel = ({
             <div className='licenseInformation'>
                 <div className='licenseInformation__Header'>
                     <span className='licenseInformation__Title'>{'License details'}</span>
-                    <button
-                        className='btn btn-primary btn-sm add-seats-button '
+                    <Button
+                        emphasis='primary'
```
Comment (larkox) [inline webapp/channels/src/components/admin_console/license_settings/enterprise_edition/enterprise_edition_left_panel.tsx:224]: Same here with the add-seats-button class.

File: webapp/channels/src/components/admin_console/license_settings/modals/upload_license_modal.tsx [hunk 20]
```diff
@@ -197,8 +198,8 @@ const UploadLicenseModal = (props: Props): JSX.Element | null => {
             </div>
             <div className='content-footer'>
                 <div className='btn-upload-wrapper'>
-                    <button
-                        className={`btn ${(fileObj?.name && fileObj?.name.length > 0) && 'btn-primary'}`}
+                    <Button
+                        emphasis='primary'
```
Comment (larkox) [inline webapp/channels/src/components/admin_console/license_settings/modals/upload_license_modal.tsx:202]: There seems to be some weird logic here to decide whether the button is primary, or something else. I guess that not having anything would either way default to primary, but still, would be nice to doublecheck.

File: webapp/channels/src/components/admin_console/plugin_management/__snapshots__/plugin_management.test.tsx.snap [hunk 21]
```diff
@@ -3576,7 +3576,7 @@ exports[`components/PluginManagement should match snapshot, upload disabled 1`]
                     class="file__upload"
                   >
                     <button
-                      class="btn"
+                      class="btn btn-tertiary"
```
Comment (larkox) [inline webapp/channels/src/components/admin_console/plugin_management/__snapshots__/plugin_management.test.tsx.snap:3579]: I saw your comment in the other file, but this used not to have the `btn-tertiary`, so I wonder if the test is not great, or we are changing some behavior here.

File: webapp/channels/src/components/admin_console/request_button/request_button.tsx [hunk 22]
```diff
@@ -209,9 +210,7 @@ export default class RequestButton extends React.PureComponent<Props, State> {
                 </div>
             );
         } else if (this.state.success && this.props.showSuccessMessage) {
-            const text = typeof this.props.successMessage === 'string' ?
-                this.props.successMessage :
-                (<FormattedMessage {...this.props.successMessage}/>);
+            const text = typeof this.props.successMessage === 'string' ? this.props.successMessage : (<FormattedMessage {...this.props.successMessage}/>);
```
Comment (larkox) [inline webapp/channels/src/components/admin_console/request_button/request_button.tsx]: Nitpick: Cursor loves to put the `?` operator in a single line...

0/5 whether you want to keep the change, or revert it just to keep the diff cleaner.

File: webapp/channels/src/components/admin_console/team_channel_settings/__snapshots__/abstract_list.test.tsx.snap [hunk 23]
```diff
@@ -120,7 +120,7 @@ exports[`admin_console/team_channel_settings/AbstractList should match snapshot,
         1 - 1 of 1
       </div>
       <button
-        class="btn btn-tertiary prev disabled"
+        class="btn btn-tertiary disabled"
```
Comment (larkox) [inline webapp/channels/src/components/admin_console/team_channel_settings/__snapshots__/abstract_list.test.tsx.snap:123]: More `prev`/`next` missing.

File: webapp/channels/src/components/admin_console/team_channel_settings/channel/details/__snapshots__/channel_details.test.tsx.snap [hunk 24]
```diff
@@ -385,7 +385,7 @@ exports[`admin_console/team_channel_settings/channel/ChannelDetails should match
                 1 - 1 of 1
               </div>
               <button
-                class="btn btn-tertiary prev disabled"
+                class="btn btn-tertiary disabled"
```
Comment (larkox) [inline webapp/channels/src/components/admin_console/team_channel_settings/channel/details/__snapshots__/channel_details.test.tsx.snap:388]: And more.

File: webapp/channels/src/components/admin_console/team_channel_settings/channel/details/__snapshots__/channel_groups.test.tsx.snap [hunk 25]
```diff
@@ -80,7 +80,7 @@ exports[`admin_console/team_channel_settings/channel/ChannelGroups should match
           1 - 1 of 1
         </div>
         <button
-          class="btn btn-tertiary prev disabled"
+          class="btn btn-tertiary disabled"
```
Comment (larkox) [inline webapp/channels/src/components/admin_console/team_channel_settings/channel/details/__snapshots__/channel_groups.test.tsx.snap:83]: And more.

File: webapp/channels/src/components/admin_console/team_channel_settings/group/__snapshots__/group_list.test.tsx.snap [hunk 26]
```diff
@@ -101,7 +101,7 @@ exports[`admin_console/team_channel_settings/group/GroupList should match snapsh
         1 - 1 of 1
       </div>
       <button
-        class="btn btn-tertiary prev disabled"
+        class="btn btn-tertiary disabled"
```
Comment (larkox) [inline webapp/channels/src/components/admin_console/team_channel_settings/group/__snapshots__/group_list.test.tsx.snap:104]: And more.

File: webapp/channels/src/components/admin_console/team_channel_settings/team/details/__snapshots__/team_groups.test.tsx.snap [hunk 27]
```diff
@@ -80,7 +80,7 @@ exports[`admin_console/team_channel_settings/team/TeamGroups should match snapsh
           1 - 1 of 1
         </div>
         <button
-          class="btn btn-tertiary prev disabled"
+          class="btn btn-tertiary disabled"
```
Comment (larkox) [inline webapp/channels/src/components/admin_console/team_channel_settings/team/details/__snapshots__/team_groups.test.tsx.snap:83]: And... last time I am going to mention the missing prev and next XD

File: webapp/channels/src/components/admin_console/team_channel_settings/team/details/__snapshots__/team_profile.test.tsx.snap [hunk 28]
```diff
@@ -82,7 +82,7 @@ exports[`admin_console/team_channel_settings/team/TeamProfile should match snaps
           class="AdminChannelDetails_archiveContainer"
         >
           <button
-            class="btn ArchiveButton ArchiveButton___unarchived cloud-limits-disabled"
+            class="btn btn-secondary btn-danger"
```
Comment (larkox) [inline webapp/channels/src/components/admin_console/team_channel_settings/team/details/__snapshots__/team_profile.test.tsx.snap:85]: Probably intended, but we are losing a few classes here that I don't see explicitly being removed from the code, so flagging it just in case.

File: webapp/channels/src/components/admin_console/workspace-optimization/__snapshots__/cta_buttons.test.tsx.snap [hunk 29]
```diff
@@ -6,12 +6,12 @@ exports[`components/admin_console/workspace-optimization/cta_buttons should matc
...
   >
     <button
-      class="btn btn-primary btn-sm actionButton annnouncementBar__purchaseNow"
+      class="btn btn-primary btn-sm annnouncementBar__purchaseNow"
     >
       Action Text
     </button>
     <button
-      class="btn btn-tertiary btn-sm learnMoreButton"
+      class="btn btn-tertiary btn-sm"
```
Comment (larkox) [inline webapp/channels/src/components/admin_console/workspace-optimization/__snapshots__/cta_buttons.test.tsx.snap:14]: Another two classes that have dissappeared.

File: webapp/channels/src/components/apps_form/__snapshots__/apps_form_container.test.tsx.snap [hunk 30]
```diff
@@ -232,14 +232,14 @@ exports[`components/apps_form/AppsFormContainer should match snapshot 1`] = `
             >
               <div>
                 <button
-                  class="btn btn-tertiary cancel-button"
+                  class="btn btn-tertiary"
```
Comment (larkox) [inline webapp/channels/src/components/apps_form/__snapshots__/apps_form_container.test.tsx.snap:235]: More missing classes.

File: webapp/channels/src/components/cloud_subscribe_result_modal/__snapshots__/success.test.tsx.snap [hunk 31]
```diff
@@ -140,7 +140,7 @@ exports[`components/pricing_modal/downgrade_team_removal_modal matches snapshot
               class="IconMessage-button"
             >
               <button
-                class="btn btn-primary Form-btn"
+                class="btn btn-primary"
```
Comment (larkox) [inline webapp/channels/src/components/cloud_subscribe_result_modal/__snapshots__/success.test.tsx.snap:143]: Another missing class.

File: webapp/channels/src/components/commercial_support_modal/commercial_support_modal.test.tsx [hunk 32]
```diff
@@ -77,7 +77,7 @@ describe('components/CommercialSupportModal', () => {
         renderWithContext(<CommercialSupportModal {...baseProps}/>);
 
         const user = userEvent.setup();
-        const downloadLink = screen.getByText('Download Support Packet').closest('a');
+        const downloadLink = screen.getByText('Download Support Packet').closest('button');
```
Comment (larkox) [inline webapp/channels/src/components/commercial_support_modal/commercial_support_modal.test.tsx]: 0/5 do we want to change this to get by role?

File: webapp/channels/src/components/edit_channel_header_modal/__snapshots__/edit_channel_header_modal.test.tsx.snap [hunk 33]
```diff
@@ -177,13 +177,13 @@ exports[`components/EditChannelHeaderModal edit direct message channel 1`] = `
             class="modal-footer"
           >
             <button
-              class="btn btn-tertiary cancel-button"
```
Comment (larkox) [inline webapp/channels/src/components/edit_channel_header_modal/__snapshots__/edit_channel_header_modal.test.tsx.snap:180]: More missing classes.

File: webapp/channels/src/components/invitation_modal/result_view.tsx [hunk 34]
```diff
@@ -88,19 +90,19 @@ export default function ResultView(props: Props) {
                 )}
             </Modal.Body>
             <Modal.Footer className={classNames('InviteView__footer', props.footerClass)}>
-                <button
+                <Button
                     onClick={props.inviteMore}
-                    className='btn btn-tertiary ResultView__inviteMore'
+                    emphasis='tertiary'
```
Comment (larkox) [inline webapp/channels/src/components/invitation_modal/result_view.tsx:95]: Another class removed.

File: webapp/channels/src/components/product_notices_modal/__snapshots__/product_notices.test.tsx.snap [hunk 35]
```diff
@@ -229,7 +228,7 @@ exports[`ProductNoticesModal Match snapshot for user notice 1`] = `
               class="modal-footer"
             >
               <button
-                class="GenericModal__button btn btn-tertiary"
+                class="btn btn-tertiary"
```
Comment (larkox) [inline webapp/channels/src/components/product_notices_modal/__snapshots__/product_notices.test.tsx.snap:231]: Missing class for GenericModal__button?

File: webapp/channels/src/components/team_members_modal/team_members_modal.tsx [hunk 36]
```diff
@@ -101,17 +102,18 @@ export default class TeamMembersModal extends React.PureComponent<Props, State>
                         teamId={this.props.currentTeam?.id}
                         permissions={[Permissions.ADD_USER_TO_TEAM, Permissions.INVITE_GUEST]}
                     >
-                        <button
+                        <Button
                             id='invitePeople'
                             type='button'
-                            className='btn btn-primary btn-sm invite-people-btn'
```
Comment (larkox) [inline webapp/channels/src/components/team_members_modal/team_members_modal.tsx:107]: More missing classes.

File: webapp/channels/src/components/view_user_group_modal/view_user_group_modal_header/view_user_group_modal_header.tsx [hunk 37]
```diff
@@ -120,15 +123,16 @@ const ViewUserGroupModalHeader = ({
     const restoreGroupButton = useCallback(() => {
         if (permissionToRestoreGroup) {
             return (
-                <button
-                    className='user-groups-create btn btn-sm btn-secondary'
```
Comment (larkox) [inline webapp/channels/src/components/view_user_group_modal/view_user_group_modal_header/view_user_group_modal_header.tsx:124]: Another missing class.

File: webapp/channels/src/components/widgets/modals/components/save_changes_panel.tsx [hunk 38]
```diff
@@ -107,9 +106,11 @@ function SaveChangesPanel({
 
         return (
             <div className='SaveChangesPanel__btn-ctr'>
-                <button
+                <Button
                     data-testid='SaveChangesPanel__cancel-btn'
-                    className={cancelButtonClassName}
+                    emphasis='tertiary'
+                    size='sm'
+                    className={classNames({error: tabChangeError || state === 'error'})}
```
Comment (larkox) [inline webapp/channels/src/components/widgets/modals/components/save_changes_panel.tsx:113]: Some classes are missing here too. In particular `saved` and `SaveChangesPanel__save-btn`

**Review by larkox — APPROVED** [review]
Most of my comments are probably addressed by your comment where you say that you removed the classes that have no styles attached. But better safe than sorry.

Another thing I would love to see in this PR is the button component to be in the Component Library, but I leave up to you whether that should be in this PR or not.

File: webapp/channels/src/components/__snapshots__/setting_picture.test.tsx.snap (hunk 16 above)
Comment (hmhealey) [reply webapp/channels/src/components/__snapshots__/setting_picture.test.tsx.snap:208]: That's fine. `btn-primary` looks the same as `btn-inactive` when it's disabled, and the `disabled` class was redundant

File: webapp/channels/src/components/admin_console/group_settings/groups_list/__snapshots__/groups_list.test.tsx.snap (hunk 17 above)
Comment (hmhealey) [reply webapp/channels/src/components/admin_console/group_settings/groups_list/__snapshots__/groups_list.test.tsx.snap:474]: Yeah, that's intended. The only place those classes have an effect is in carousels which isn't what any of the affected components are

File: webapp/channels/src/components/admin_console/license_settings/enterprise_edition/enterprise_edition_left_panel.tsx (hunk 18 above)
Comment (hmhealey) [reply webapp/channels/src/components/admin_console/license_settings/enterprise_edition/enterprise_edition_left_panel.tsx:166]: Yep, that class doesn't have any CSS rules applied

File: webapp/channels/src/components/admin_console/license_settings/modals/upload_license_modal.tsx (hunk 20 above)
Comment (hmhealey) [reply webapp/channels/src/components/admin_console/license_settings/modals/upload_license_modal.tsx:202]: That's similar to the case in `setting_picture.test.tsx.snap` where we were either:
1. When the button is disabled, we give it just the `.btn` class
2. When the button is enabled, we give it the `.btn` and `.btn-primary` classes

That isn't needed though because the disabled styling takes precedence

File: webapp/channels/src/components/admin_console/plugin_management/__snapshots__/plugin_management.test.tsx.snap (hunk 21 above)
Comment (hmhealey) [reply webapp/channels/src/components/admin_console/plugin_management/__snapshots__/plugin_management.test.tsx.snap:3579]: That's another case where the disabled style takes precedence

File: webapp/channels/src/components/admin_console/request_button/request_button.tsx (hunk 22 above)
Comment (hmhealey) [reply webapp/channels/src/components/admin_console/request_button/request_button.tsx]: The reason Cursor does that (and I did that here) is because `eslint --fix` changes that. I tried to revert that wherever I saw that, but I forgot that here.

Personally, I don't like long ternary statements anyway, so maybe we need to have a long bikeshedding discussion on changing this rule in Web Guild :P

I'll revert it here

File: webapp/channels/src/components/admin_console/team_channel_settings/team/details/__snapshots__/team_profile.test.tsx.snap (hunk 28 above)
Comment (hmhealey) [reply webapp/channels/src/components/admin_console/team_channel_settings/team/details/__snapshots__/team_profile.test.tsx.snap:85]: Yeah, those are the ones removed from `_groups.scss` since they duplicated the and `team_profile.scss` since they weren't needed. I tested these manually

File: webapp/channels/src/components/admin_console/workspace-optimization/__snapshots__/cta_buttons.test.tsx.snap (hunk 29 above)
Comment (hmhealey) [reply webapp/channels/src/components/admin_console/workspace-optimization/__snapshots__/cta_buttons.test.tsx.snap:14]: And another unused one

File: webapp/channels/src/components/apps_form/__snapshots__/apps_form_container.test.tsx.snap (hunk 30 above)
Comment (hmhealey) [reply webapp/channels/src/components/apps_form/__snapshots__/apps_form_container.test.tsx.snap:235]: That one's used in the system console but has no effect here

File: webapp/channels/src/components/cloud_subscribe_result_modal/__snapshots__/success.test.tsx.snap (hunk 31 above)
Comment (hmhealey) [reply webapp/channels/src/components/cloud_subscribe_result_modal/__snapshots__/success.test.tsx.snap:143]: No CSS applied

File: webapp/channels/src/components/edit_channel_header_modal/__snapshots__/edit_channel_header_modal.test.tsx.snap (hunk 33 above)
Comment (hmhealey) [reply webapp/channels/src/components/edit_channel_header_modal/__snapshots__/edit_channel_header_modal.test.tsx.snap:180]: Unused class outside system console

File: webapp/channels/src/components/invitation_modal/result_view.tsx (hunk 34 above)
Comment (hmhealey) [reply webapp/channels/src/components/invitation_modal/result_view.tsx:95]: Class with no styles

File: webapp/channels/src/components/product_notices_modal/__snapshots__/product_notices.test.tsx.snap (hunk 35 above)
Comment (hmhealey) [reply webapp/channels/src/components/product_notices_modal/__snapshots__/product_notices.test.tsx.snap:231]: That class has no CSS applied. I would've removed it everywhere, but it was used in some E2E tests. In this case, it didn't affect any E2E tests though

File: webapp/channels/src/components/team_members_modal/team_members_modal.tsx (hunk 36 above)
Comment (hmhealey) [reply webapp/channels/src/components/team_members_modal/team_members_modal.tsx:107]: No CSS

File: webapp/channels/src/components/view_user_group_modal/view_user_group_modal_header/view_user_group_modal_header.tsx (hunk 37 above)
Comment (hmhealey) [reply webapp/channels/src/components/view_user_group_modal/view_user_group_modal_header/view_user_group_modal_header.tsx:124]: No CSS

File: webapp/channels/src/components/widgets/modals/components/save_changes_panel.tsx (hunk 38 above)
Comment (hmhealey) [reply webapp/channels/src/components/widgets/modals/components/save_changes_panel.tsx:113]: The `SaveChangesPanel__*` classes aren't needed. They're duplicating test IDs. I don't think the `saved` class does anything, but I'm less confident in that, so I'll re-add that

Comment (hmhealey) [reply webapp/channels/src/components/widgets/modals/components/save_changes_panel.tsx:113]: Oh, I remember now. The `saved` classes on the buttons don't have CSS attached. Those classes do have CSS for outer elements, but they don't need to be applied to the button too

File: webapp/platform/shared/src/components/button/button.tsx (hunk 15 above)
Comment (hmhealey) [reply webapp/platform/shared/src/components/button/button.tsx]: I actually decided to remove this because we only use `btn-full` in one place currently, and it's better to do this by using `display: flex` on a parent element instead of changing the Button's `width` style
